lessons/modelevaluation.ipynb

```diff
@@ -0,0 +1,786 @@
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Bayesian Model Comparison\n",
+    "\n",
+    "Inference on parameters $\\theta$ given model $H$:\n",
+    "\n",
+    "$P(\\theta|D,H)=\\frac{P(D|\\theta,H)P(\\theta|H)}{P(D|H)}$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "Inference on models $H$: \n",
+    "\n",
+    "$P(H|D,\\Omega)=\\frac{P(D|H,\\Omega)P(H|\\Omega)}{P(D|\\Omega)}$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Here $\\Omega$ is some space of all allowed models. As we normally do for parameter inference, we'll work with a simplified version:\n",
+    "\n",
+    "$P(H|D)\\propto P(D|H)P(H)$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "$P(H)$ is a prior on the model, and\n",
+    "\n",
+    "$P(D|H)=\\int P(D|\\theta,H) \\, P(\\theta|H) d\\theta$ \n",
+    "\n",
+    "is the **evidence** - the normalizing denominator in Bayesian parameter inference (also known as the **fully marginalized likelihood**)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Ideally, we would compare models by looking at\n",
+    "\n",
+    "$\\frac{P(H_2|D)}{P(H_1|D)}=\\frac{P(D|H_2)\\,P(H_2)}{P(D|H_1)\\,P(H_1)}$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "General difficulties in computing the terms in this ratio:\n",
+    "* Assigning meaningful priors to models\n",
+    "* Assigning meaningful priors to parameters\n",
+    "* Calculating the evidence integral"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "**Exercise: \"non-informative\" parameter priors and the evidence**\n",
+    "\n",
+    "Say we have a model with 1 parameter, $\\theta$, and a likelihood that works out to be a unit width Gaussian centered on $\\theta=0$ with peak value $\\mathcal{L}_{\\rm max}$.\n",
+    "\n",
+    "For each of the priors on $\\theta$ below, (a) sketch the likelihood and prior as a function of theta, (b) calculate the evidence (well enough to comment about the relative differences, anyway).\n",
+    "1. $P(\\theta)$ uniform on $[-1,+1]$\n",
+    "2. $P(\\theta)$ uniform on $[-3,+3]$\n",
+    "3. $P(\\theta)$ uniform on $[-100,+100]$\n",
+    "4. $P(\\theta)$ uniform on $[-\\infty,+\\infty]$\n",
+    "5. $P(\\theta)$ uniform on $[+3,+5]$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "**Evidence exercise: notes**\n",
```

Comment:
Somewhere here it would be good to explicitly note that the choice of priors counts as part of "the model".

Reply:
Yup - added